WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerActivity.java

```diff
@@ -310,13 +325,40 @@ public void doNext(Uri uri) {
                         }
                     });
         } else {
-            Intent intent = new Intent()
-                    .putExtra(MediaPickerConstants.EXTRA_MEDIA_URIS, convertUrisListToStringArray(mediaUris))
-                    .putExtra(MediaPickerConstants.EXTRA_MEDIA_SOURCE, source.name())
-                    // set the browserType in the result, so caller can distinguish and handle things as needed
-                    .putExtra(ARG_BROWSER_TYPE, mBrowserType);
-            setResult(RESULT_OK, intent);
-            finish();
+            // setup progress dialog
+            VisibleProgressDialog visibleProgressDialog = new VisibleProgressDialog(
+                    new UiStringRes(R.string.add_media_progress),
+                    false,
+                    true
+            );
+            mAddingMediaToEditorProgressDialog = mProgressDialogHelper
+                    .updateProgressDialogState(this, mAddingMediaToEditorProgressDialog, visibleProgressDialog,
+                            mUiHelpers);
+
+            new Timer().schedule(new TimerTask() {
```

Comment:
This needs to be run after a delay to allow the UI to get updated with the progress dialog, for which we could use `Handler().postDelayed`, but  we  need it not run in the UI thread, hence we're using a java `Thread`, scheduled to run with a `Timer`/ `TimerTask`. We could also have used AsyncTask but it's deprecated. The best option would be to use Kotlin Coroutines for this, probably a good fit for the picker rewrite effort.

Reply:
There are coroutines used in the PhotoPickerFragment and the PhotoPickerViewModel. Since we'll still be using this old photo picker for the stories, do you think it would make sense to use them?